### PR TITLE
CRM457-1112 Remove Authentication for local development

### PR DESCRIPTION
**Pull request by jogilder**
## Description of change
Have overridden the creation of `SingleTenantAzureAuthorizationCodeBearer` so that it will be ignored if the environment variable `AUTHENTICATION_REQUIRED` is set to anything other than True

Have also replaced start_up events with lifespan as they have been deprecated.

New tests to cover the unauthenticated situation

Updated envVars so can have different requirements for testing than dev if required.

## Link to relevant ticket
[CRM457-1152](https://dsdmoj.atlassian.net/browse/CRM457-1152)

## How to manually test the feature
Download this branch and point a provider application at it. This should work without specifying azure authentication tokens in the provider application.

You should also look to try and see if authentication works locally if environment variable set to True

[CRM457-1152]: https://dsdmoj.atlassian.net/browse/CRM457-1152?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by patrick-laa — COMMENTED** [review]
I don't know python so this is all probably nonsense.

File: laa_crime_application_store_app/main.py [hunk 1]
```diff
@@ -90,15 +97,13 @@ async def validation_exception_handler(request, exc):
...
+@asynccontextmanager
+async def lifespan():
     """
-    Load OpenID config on startup.
+    Load OpenID and Alembic configs on startup.
     """
-    await azure_auth.openid_config.load_config()
+    alembic_cfg = Config(f"{os.getcwd()}/alembic.ini")
+    upgrade(alembic_cfg, "head")
+    if get_auth_settings().authentication_required.lower() != "true":
```
[the remark this answers is not in the document]
Comment (jogilder) [reply laa_crime_application_store_app/main.py:107]: No, you are not being dim but it shows I don't need the if statement, I will update this.

File: .env.sample [hunk 2]
```diff
@@ -5,8 +5,10 @@ POSTGRES_HOSTNAME=localhost
...
 
 # Authentication requirements
+AUTHENTICATION_REQUIRED=TRUE
 APP_CLIENT_ID=
 TENANT_ID=
 
-# Required to display swagger docs
-SWAGGER_ENDPOINT=/docs
+# Required to display swagger docs locally
+# When set locally test_main_route_returns_404_when_disabled test will fail
```
[the remark this answers is not in the document]
Comment (ivanELEC) [reply .env.sample:13]: I don't know that there's an in built way but this package would do this: https://pypi.org/project/pytest-dotenv/

We'd just have to define a specific env file and override existing values in config

File: helm_deploy/laa-crime-application-store/templates/deployment.yaml [hunk 3]
```diff
@@ -85,6 +85,8 @@ spec:
                   key: tenant_id
             - name: SWAGGER_ENDPOINT
               value: {{ .Values.secrets.swaggerEndpoint | quote}}
+            - name: AUTHENTICATION_REQUIRED
+              value: {{ .Values.secrets.authenticationRequired | quote}}
```
[the remark this answers is not in the document]
Comment (ivanELEC) [reply helm_deploy/laa-crime-application-store/templates/deployment.yaml:89]: This is a miscommunication we have in all our apps tbh - we have what are essentially just variables/configurations in helm that are namespaced as secrets. Could clean it up here or chalk it up as tech debt

File: .env.sample (hunk 2 above)
[the remark this answers is not in the document]
Comment (jogilder) [reply .env.sample:13]: I will look into this. It is not as simple for pipenv unfortunately.

File: helm_deploy/laa-crime-application-store/templates/deployment.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jogilder) [reply helm_deploy/laa-crime-application-store/templates/deployment.yaml:89]: I don't know why it is called secrets they are in fact env Vars. I can try renaming.

[the remark this answers is not in the document]
Comment (jogilder) [reply helm_deploy/laa-crime-application-store/templates/deployment.yaml]: Renamed as envVars and only allow change if not in prod

File: .env.sample (hunk 2 above)
[the remark this answers is not in the document]
Comment (jogilder) [reply .env.sample]: Have updated and added in extra env example files

**Review by patrick-laa — APPROVED** [review]
LGTM but there are now some merge conflicts to resolve.

**Pull request merged**